sync/trypush.py

```diff
@@ -350,19 +351,44 @@ def stability(self):
         """Is the current try push a stability test"""
         return self["stability"]
 
+    @property
+    def bug(self):
+        return self.get("bug")
+
     @property
     def infra_fail(self):
-        """Is the current try push a stability test"""
+        """Does this push have infrastructure failures"""
         if self.status == "infra-fail":
             self.status = "complete"
             self.infra_fail = True
         return self.get("infra-fail", False)
 
+    def notify_failed_builds(self):
+        tasks = tc.TaskGroup(self.taskgroup_id).view()
+        failed = tasks.failed_builds()
+
+        if not failed:
+            logger.error("No failed builds to report for Try push: %s" % self)
+            return
+
+        bug = self.bug
+
+        if not bug:
+            logger.error("No associated bug found for Try push: %s" % self)
+            return
+
+        msg = "There were infrastructure failures for the Try push (%s):\n" % self.treeherder_url
+        msg += "\n".join(task.get("task", {}).get("metadata", {}).get("name")
+                         for task in failed.tasks)
+        env.bz.comment(self.bug, msg)
+
     @infra_fail.setter
     @mut()
     def infra_fail(self, value):
-        """Is the current try push a stability test"""
+        """Set the status of this push's infrastructure failure state"""
         self["infra-fail"] = value
```

Comment:
Can we return if the value isn't changing?

Reply:
Would be a good idea, will implement that.